File: cosmopratiche/src/java/it/csi/cosmo/cosmopratiche/business/service/TabsDettaglioService.java
/*
 * Copyright CSI-Piemonte - 2023
 * SPDX-License-Identifier: GPL-3.0-or-later
 */

package it.csi.cosmo.cosmopratiche.business.service;
import java.util.List;

import it.csi.cosmo.common.entities.CosmoRTabDettaglioTipoPratica;
import it.csi.cosmo.cosmopratiche.dto.rest.TabsDettaglio;

public interface TabsDettaglioService {

  /**
   * Metodo che restituisce una lista di Tabs
   *
   * @return una lista di tabs 
   */
	List<TabsDettaglio> getTabsDettaglio();

	List<TabsDettaglio> getTabsDettaglioCodiceTipoPratica(String codiceTipoPratica);

  
}

File: cosmowcl/src/app/shared/services/tabs-dettaglio.service.ts
/*
 * Copyright CSI-Piemonte - 2023
 * SPDX-License-Identifier: GPL-3.0-or-later
 */
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { ApiUrls } from '../utilities/apiurls';
import { Observable } from 'rxjs';
import { TabsDettaglio } from '../models/api/cosmopratiche/tabsDettaglio';
import { Utils } from '../utilities/utilities';

@Injectable({
  providedIn: 'root'
})
export class TabsDettaglioService {

  constructor(
    private http: HttpClient
  ) { }

  getTabsDettaglio(): Observable<TabsDettaglio[]> {
    return this.http.get<TabsDettaglio[]>(ApiUrls.TABS_DETTAGLIO);
  }

  getTabsDettaglioCodiceTipoPratica(codiceTipoPratica: string): Observable<TabsDettaglio[]> {
    Utils.require(codiceTipoPratica);
    return this.http.get<TabsDettaglio[]>(ApiUrls.TABS_DETTAGLIO_FROM_CODICE_TIPO_PRATICA
      .replace('{codiceTipoPratica}', codiceTipoPratica.toString()));
  }

}
